pkg/util/archive/zip.go

```diff
@@ -0,0 +1,261 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package archive provides functions to archive and unarchive files.
+package archive
+
+import (
+	"archive/zip"
+	"compress/flate"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// Zip is an adapted implementation of (*Zip).Archive from
+// https://github.com/mholt/archiver/blob/v3.5.1/zip.go#L140
+// Copyright (c) 2016 Matthew Holt
+func Zip(sources []string, destination string) error {
+	if !strings.HasSuffix(destination, ".zip") {
+		return fmt.Errorf("%s must have a .zip extension", destination)
+	}
+	if fileExists(destination) {
+		return fmt.Errorf("file already exists: %s", destination)
+	}
+	dir := filepath.Dir(destination)
+	if !fileExists(dir) {
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			return fmt.Errorf("%w making directory: %s", err, dir)
+		}
+	}
+
+	out, err := os.Create(destination)
+	if err != nil {
+		return fmt.Errorf("%w creating %s", err, destination)
+	}
+	defer out.Close()
+
+	zipW := zip.NewWriter(out)
+	zipW.RegisterCompressor(zip.Deflate, func(out io.Writer) (io.WriteCloser, error) {
+		return flate.NewWriter(out, flate.DefaultCompression)
```

Comment:
Big question here - were the flares compressed using custom/optimized `Deflate` [through a dependency](https://github.com/klauspost/compress?tab=readme-ov-file#deflate-usage)? If so, we may be regressing in speed and file size in the output so it would be good to do at least a side-by-side size comparison. The changeover is likely acceptable but we should at least understand the impact.

Reply:
That is a great point. 

There are two side to that quetsion:

Reply:
> having a comparison table would be useful to answer the questions above
I think just a simple flare from both implementations should be fine and we can just check the file size.

For your questions:
- Do we want to reduce the number of imports? Yes
- Are performance degradation an issue for creating flares? No, as long as they're not drastic

Reply:
<img width="843" alt="image" src="https://github.com/DataDog/datadog-agent/assets/4672858/7cc9dd43-5e7b-43bb-8737-13d1294caa40">
main_branch -> main
custom_branch|custom_version -> this branch 

Both archive are the same size. Uncompressed custom branch is 0.1 MB smaller.

I think we good with the change 😄